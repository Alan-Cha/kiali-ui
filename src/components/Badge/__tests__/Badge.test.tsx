import * as React from 'react';
import { shallow } from 'enzyme';
import Badge from '../Badge';

describe('#Badge render correctly with data', () => {
  it('should render badge', () => {
    const wrapper = shallow(<Badge scale={0.8} style="plastic" color="green" leftText="my_key" rightText="my_value" />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
