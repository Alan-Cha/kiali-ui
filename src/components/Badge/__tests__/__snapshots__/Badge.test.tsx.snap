// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Badge render correctly with data should render badge 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Badge
    color="green"
    leftText="my_key"
    rightText="my_value"
    scale={0.8}
    style="plastic"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <linearGradient
          id="a"
          x2="0"
          y2="100%"
>
          <stop
                    offset="0"
                    stopColor="#bbb"
                    stopOpacity=".1"
          />
          <stop
                    offset="1"
                    stopOpacity=".1"
          />
</linearGradient>,
        <rect
          fill="#555"
          height={16}
          rx={2.4000000000000004}
          width={19.2}
/>,
        <rect
          fill="green"
          height={16}
          rx={2.4000000000000004}
          width={9.6}
          x={9.6}
/>,
        <rect
          fill="green"
          height={16}
          width="13"
          x={9.6}
/>,
        <rect
          fill="url(#a)"
          height={16}
          rx={2.4000000000000004}
          width={19.2}
/>,
        <g
          fill="#fff"
          fontFamily="DejaVu Sans,Verdana,Geneva,sans-serif"
          fontSize={8.8}
>
          <text
                    fill="#010101"
                    fillOpacity=".3"
                    x={4.8}
                    y={12}
          >
                    my_key
          </text>
          <text
                    x={4.8}
                    y={11.2}
          >
                    my_key
          </text>
          <text
                    fill="#010101"
                    fillOpacity=".3"
                    x={14.399999999999999}
                    y={12}
          >
                    my_value
          </text>
          <text
                    x={14.399999999999999}
                    y={11.2}
          >
                    my_value
          </text>
</g>,
      ],
      "height": 16,
      "preserveAspectRatio": "xMidYMid meet",
      "style": Object {
        "marginLeft": "5px",
      },
      "version": "1.0",
      "width": 19.2,
      "xmlns": "http://www.w3.org/2000/svg",
      "zoomAndPan": "magnify",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <stop
              offset="0"
              stopColor="#bbb"
              stopOpacity=".1"
/>,
            <stop
              offset="1"
              stopOpacity=".1"
/>,
          ],
          "id": "a",
          "x2": "0",
          "y2": "100%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "offset": "0",
              "stopColor": "#bbb",
              "stopOpacity": ".1",
            },
            "ref": null,
            "rendered": null,
            "type": "stop",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "offset": "1",
              "stopOpacity": ".1",
            },
            "ref": null,
            "rendered": null,
            "type": "stop",
          },
        ],
        "type": "linearGradient",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "fill": "#555",
          "height": 16,
          "rx": 2.4000000000000004,
          "width": 19.2,
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "fill": "green",
          "height": 16,
          "rx": 2.4000000000000004,
          "width": 9.6,
          "x": 9.6,
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "fill": "green",
          "height": 16,
          "width": "13",
          "x": 9.6,
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "fill": "url(#a)",
          "height": 16,
          "rx": 2.4000000000000004,
          "width": 19.2,
        },
        "ref": null,
        "rendered": null,
        "type": "rect",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <text
              fill="#010101"
              fillOpacity=".3"
              x={4.8}
              y={12}
>
              my_key
</text>,
            <text
              x={4.8}
              y={11.2}
>
              my_key
</text>,
            <text
              fill="#010101"
              fillOpacity=".3"
              x={14.399999999999999}
              y={12}
>
              my_value
</text>,
            <text
              x={14.399999999999999}
              y={11.2}
>
              my_value
</text>,
          ],
          "fill": "#fff",
          "fontFamily": "DejaVu Sans,Verdana,Geneva,sans-serif",
          "fontSize": 8.8,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "my_key",
              "fill": "#010101",
              "fillOpacity": ".3",
              "x": 4.8,
              "y": 12,
            },
            "ref": [Function],
            "rendered": "my_key",
            "type": "text",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "my_key",
              "x": 4.8,
              "y": 11.2,
            },
            "ref": null,
            "rendered": "my_key",
            "type": "text",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "my_value",
              "fill": "#010101",
              "fillOpacity": ".3",
              "x": 14.399999999999999,
              "y": 12,
            },
            "ref": [Function],
            "rendered": "my_value",
            "type": "text",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "my_value",
              "x": 14.399999999999999,
              "y": 11.2,
            },
            "ref": null,
            "rendered": "my_value",
            "type": "text",
          },
        ],
        "type": "g",
      },
    ],
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <linearGradient
            id="a"
            x2="0"
            y2="100%"
>
            <stop
                        offset="0"
                        stopColor="#bbb"
                        stopOpacity=".1"
            />
            <stop
                        offset="1"
                        stopOpacity=".1"
            />
</linearGradient>,
          <rect
            fill="#555"
            height={16}
            rx={2.4000000000000004}
            width={19.2}
/>,
          <rect
            fill="green"
            height={16}
            rx={2.4000000000000004}
            width={9.6}
            x={9.6}
/>,
          <rect
            fill="green"
            height={16}
            width="13"
            x={9.6}
/>,
          <rect
            fill="url(#a)"
            height={16}
            rx={2.4000000000000004}
            width={19.2}
/>,
          <g
            fill="#fff"
            fontFamily="DejaVu Sans,Verdana,Geneva,sans-serif"
            fontSize={8.8}
>
            <text
                        fill="#010101"
                        fillOpacity=".3"
                        x={4.8}
                        y={12}
            >
                        my_key
            </text>
            <text
                        x={4.8}
                        y={11.2}
            >
                        my_key
            </text>
            <text
                        fill="#010101"
                        fillOpacity=".3"
                        x={14.399999999999999}
                        y={12}
            >
                        my_value
            </text>
            <text
                        x={14.399999999999999}
                        y={11.2}
            >
                        my_value
            </text>
</g>,
        ],
        "height": 16,
        "preserveAspectRatio": "xMidYMid meet",
        "style": Object {
          "marginLeft": "5px",
        },
        "version": "1.0",
        "width": 19.2,
        "xmlns": "http://www.w3.org/2000/svg",
        "zoomAndPan": "magnify",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <stop
                offset="0"
                stopColor="#bbb"
                stopOpacity=".1"
/>,
              <stop
                offset="1"
                stopOpacity=".1"
/>,
            ],
            "id": "a",
            "x2": "0",
            "y2": "100%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "offset": "0",
                "stopColor": "#bbb",
                "stopOpacity": ".1",
              },
              "ref": null,
              "rendered": null,
              "type": "stop",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "offset": "1",
                "stopOpacity": ".1",
              },
              "ref": null,
              "rendered": null,
              "type": "stop",
            },
          ],
          "type": "linearGradient",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "fill": "#555",
            "height": 16,
            "rx": 2.4000000000000004,
            "width": 19.2,
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "fill": "green",
            "height": 16,
            "rx": 2.4000000000000004,
            "width": 9.6,
            "x": 9.6,
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "fill": "green",
            "height": 16,
            "width": "13",
            "x": 9.6,
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "fill": "url(#a)",
            "height": 16,
            "rx": 2.4000000000000004,
            "width": 19.2,
          },
          "ref": null,
          "rendered": null,
          "type": "rect",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <text
                fill="#010101"
                fillOpacity=".3"
                x={4.8}
                y={12}
>
                my_key
</text>,
              <text
                x={4.8}
                y={11.2}
>
                my_key
</text>,
              <text
                fill="#010101"
                fillOpacity=".3"
                x={14.399999999999999}
                y={12}
>
                my_value
</text>,
              <text
                x={14.399999999999999}
                y={11.2}
>
                my_value
</text>,
            ],
            "fill": "#fff",
            "fontFamily": "DejaVu Sans,Verdana,Geneva,sans-serif",
            "fontSize": 8.8,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "my_key",
                "fill": "#010101",
                "fillOpacity": ".3",
                "x": 4.8,
                "y": 12,
              },
              "ref": [Function],
              "rendered": "my_key",
              "type": "text",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "my_key",
                "x": 4.8,
                "y": 11.2,
              },
              "ref": null,
              "rendered": "my_key",
              "type": "text",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "my_value",
                "fill": "#010101",
                "fillOpacity": ".3",
                "x": 14.399999999999999,
                "y": 12,
              },
              "ref": [Function],
              "rendered": "my_value",
              "type": "text",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "my_value",
                "x": 14.399999999999999,
                "y": 11.2,
              },
              "ref": null,
              "rendered": "my_value",
              "type": "text",
            },
          ],
          "type": "g",
        },
      ],
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
